staging/src/k8s.io/client-go/tools/remotecommand/remotecommand_test.go

```diff
@@ -264,3 +297,75 @@ func v4WriteStatusFunc(stream io.Writer) func(status *apierrors.StatusError) err
 		return err
 	}
 }
+
+// A writer wrap which provides a helper method to block until written.
+type writerWrap struct {
+	written chan bool
+	closed  bool
+	io.Writer
+}
+
+func newWriterWrap(w io.Writer) *writerWrap {
+	return &writerWrap{
+		written: make(chan bool),
+		Writer:  w,
+	}
+}
+
+func (w *writerWrap) BlockUntilWritten() {
+	<-w.written
+}
+
+func (w *writerWrap) Write(p []byte) (n int, err error) {
+	if !w.closed {
+		w.written <- true
```

Comment:
nit: no need to send `true`. Just close `w.written` would unblock.

Reply:
Thx, just removed.